node/bindnode/repr.go

```diff
@@ -537,15 +542,24 @@ type _assemblerRepr struct {
 	nullable bool
 }
 
-func assemblerRepr(am datamodel.NodeAssembler) *_assemblerRepr {
-	return (*_assemblerRepr)(am.(*_assembler))
+func assemblerRepr(am datamodel.NodeAssembler) datamodel.NodeAssembler {
+	switch am := am.(type) {
+	case *_assembler:
+		return (*_assemblerRepr)(am)
+	case _errorAssembler:
+		return am
+	default:
+		panic(fmt.Sprintf("unexpected NodeAssembler type: %T", am))
```

Comment:
I assume that this means that you don't see a valid path to reach here in the current code? it's got to be either an `*_assembler`, or an `_errorAssembler` or someone's done something wrong and messed with the internals somehow?

Reply:
Correct. And if we get there, I have basically no other option but to fatally fail.

I guess I could use an errorAssembler, but I also want these unexpected paths caused by internal bugs to loudly fail, because they need to be fixed. The way to ensure that a panic doesn't happen isn't by removing the panic, but rather by testing and fuzzing well :)